_posts/2019-10-15-dependency-management.md

```diff
@@ -0,0 +1,177 @@
+
+```bash
+$ coursier resolve \
+    --strict \
+    --strict-exclude 'org.scala-lang:*' \
+    --force-version org.typelevel:cats-core_2.12:1.5.0 \
+    org.typelevel:cats-effect_2.12:2.0.0 \
+    org.typelevel:cats-core_2.12:1.5.0
+org.scala-lang:scala-library:2.12.9:default
+org.scala-lang:scala-reflect:2.12.6:default
+org.typelevel:cats-core_2.12:1.5.0:default
+org.typelevel:cats-effect_2.12:2.0.0:default
+org.typelevel:cats-kernel_2.12:1.5.0:default
+org.typelevel:cats-macros_2.12:1.5.0:default
+org.typelevel:machinist_2.12:0.6.6:default
+```
+
+### sbt
+
+From sbt, enable the strict conflict manager either through the original `conflictManager` key, like
+
+```scala
+conflictManager := ConflictManager.strict
+```
+
+or enable it in a possibly more thin-grained way if [sbt-coursier](https://get-coursier.io/docs/sbt-coursier) is around with
+
+```scala
+versionReconciliation += "*" % "*" % "strict"
+```
+
+(Note that to use sbt-coursier from an sbt 1.3.x project, the [coursier-based sbt launcher](https://github.com/coursier/sbt-launcher) is required.
+Get it via its [custom sbt-extras launcher](https://github.com/coursier/sbt-extras/blob/master/sbt) or generate and run a launcher via coursier with `coursier bootstrap sbt-launcher && ./sbt`.)
+
+`update` will then succeed only if the strict checks pass.
+If they don't, either force the version of faulty dependencies with `dependencyOverrides`, like
+
+```scala
+dependencyOverrides += "org.typelevel" % "cats-core_2.12" % "1.5.0"
+```
+
+or adjust their version reconciliation, like
+
+```scala
+versionReconciliation ++= Seq(
+  "org.typelevel" %% "cats-core" % "relaxed", // "semver" reconciliation is also available
+  "*" % "*" % "strict"
+)
+```
+
+A number of version reconciliation types are available:
+
+* `"strict"` requires all dependees to depend on the exact selected version (if they depend on an interval, the selected version only needs to be contained in it)
+* `"semver"` only requires all dependees to depend on the same major version as the selected version (if they depend on an interval, the selected version only needs to be contained in it too)
+* `"default"` is the default version reconciliation in coursier, enabled in the coursier CLI or with the sbt-coursier plugin.
+  It is described in more detail [in this page](https://get-coursier.io/docs/other-version-handling.html#reconciliation).
+* `"relaxed"` does not trigger any conflict.
+  It is the default in the coursier support of sbt `1.3.x`, which allows for better compatibility with former sbt versions.
+  It ignores the lowest versions and version intervals, until they can be reconciled with the same algorithm as `"default"`.
+
+Since reconciliation strategies can be set per artifact, or per organization, it becomes much easier to declaratively tell coursier, for example, that some project adheres to SemVer, and that it is fine to resolve dependencies on that specific project according to SemVer, but not others.
```

Comment:
```suggestion
Since reconciliation strategies can be set per artifact, or per organization, it becomes much easier to declaratively tell sbt, for example, that some project adheres to SemVer, and that it is fine to resolve dependencies on that specific project according to SemVer, but not others.
```
?